lib/evaluation/markForSubmissionEvaluationHandler.ts

```diff
@@ -0,0 +1,114 @@
+import { Descriptor, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+import { nanoid } from 'nanoid';
+
+import { Status } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { EvaluationClient } from './evaluationClient';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class MarkForSubmissionEvaluationHandler extends AbstractEvaluationHandler {
+  constructor(client: EvaluationClient) {
+    super(client);
+    this.verifiablePresentation.presentationSubmission = {};
+    this.verifiablePresentation.presentationSubmission.descriptor_map = [];
+    this.verifiablePresentation.presentationSubmission.id = nanoid();
+  }
+
+  public getName(): string {
+    return 'MarkForSubmissionEvaluation';
+  }
+
+  public handle(pd: PresentationDefinition, p: any): void {
+    this.iterateOverInputCandidates(pd, p);
+  }
+
+  //TODO move to utils
+  private iterateOverInputCandidates(pd: PresentationDefinition, inputCandidates: any): void {
+    const props = Object.entries(inputCandidates).filter(
```

Comment:
props has a different connotationin the world of JS frameworks. It may be better to rename it to our domain related concepts. e.g. `verifiableCredentials` may be?

Reply:
renamed